lib/util.js

```diff
@@ -96,10 +96,36 @@ function isValidNumber(value) {
     return (typeof value === "number") && !isNaN(value);
 }
 
+/**
+ * Converts one of several potential values to a local file path.
+ * If the value is a string and does _not_ begin with "file://", then
+ * the value will be returned as-is. Otherwise the value is assumed to
+ * be a file URL, and the method will return the local path represented
+ * by the URL.
+ * @param {URL|string} path Value whose path should be retrieved.
+ * @returns {string} A local file path.
+ */
+function fileUrlToFilePath(path) {
+    if ((typeof path === "string") && !path.startsWith("file://")) {
+        return path;
+    } else {
+        const url = new URL(path);
+        let filePath = decodeURIComponent(url.pathname);
+        // windows paths will have a forward slash followed by
+        // the drive letter. strip off the leading forward slash
+        // if it appears to be a windows path (i.e. starts with "/C:/")
+        if (/^\/[a-zA-Z]:\//g.exec(filePath)) {
```

Comment:
Why is the leading forward slash here to start with?

Reply:
@tmathern [fileUrl](https://github.com/adobe/node-httptransfer/blob/8f6d85cc14046a184a4adfa85719d6e61836151b/lib/aemupload.js#L43) adds it.

Reply:
Apparently it's a requirement for file:// urls. Fwiw - we are depending on an old version of file-url to create the file:// urls. The more recent versions (v3 and v4) are depending on new NodeJS feature, which is odd because new NodeJS deployments have this functionality available in the `url` module.